app/src/main/java/org/schabi/newpipe/local/feed/FeedViewModel.kt

```diff
@@ -111,20 +119,32 @@ class FeedViewModel(
         val t4: OffsetDateTime?
     )
 
-    fun togglePlayedItems(showPlayedItems: Boolean) {
-        toggleShowPlayedItems.onNext(showPlayedItems)
+    fun setShowPlayedItems(showPlayedItems: Boolean) {
+        this.showPlayedItems.onNext(showPlayedItems)
     }
 
     fun saveShowPlayedItemsToPreferences(showPlayedItems: Boolean) =
         PreferenceManager.getDefaultSharedPreferences(application).edit {
-            this.putBoolean(application.getString(R.string.feed_show_played_items_key), showPlayedItems)
+            this.putBoolean(application.getString(R.string.feed_show_watched_items_key), showPlayedItems)
             this.apply()
         }
 
     fun getShowPlayedItemsFromPreferences() = getShowPlayedItemsFromPreferences(application)
 
-    fun toggleFutureItems(showFutureItems: Boolean) {
-        toggleShowFutureItems.onNext(showFutureItems)
+    fun setShowPartiallyPlayedItems(showPartiallyPlayedItems: Boolean) {
+        this.showPartiallyPlayedItems.onNext(showPartiallyPlayedItems)
+    }
+
+    fun saveShowPartiallyPlayedItemsToPreferences(showPartiallyPlayedItems: Boolean) =
+        PreferenceManager.getDefaultSharedPreferences(application).edit {
+            this.putBoolean(application.getString(R.string.feed_show_partially_watched_items_key), showPartiallyPlayedItems)
+            this.apply()
+        }
```

Comment:
These pairs of functions (also the other two ones) could probably be merged into one (e.g. `setSaveShowPartiallyPlayedItems`)